nemo/tron/converter/common.py

```diff
@@ -0,0 +1,346 @@
+                num_floating_point_operations_so_far=0,
+            )
+
+        print(f"Converted {self.input_path} to {self.output_path} in {source.dtype}.")
+
+        return self.output_path
+
+
+class BaseExporter:
+    def __init__(self, input_path: str | Path, output_path: str | Path, hf_tokenizer_path: Optional[str] = None):
+        self.input_path = Path(input_path) if isinstance(input_path, str) else input_path
+        self.output_path = Path(output_path) if isinstance(output_path, str) else output_path
+        self._hf_config = None
+        self._tron_config = None
+        self._hf_tokenizer_path = hf_tokenizer_path
+        self._tokenizer = None
+
+    @property
+    def hf_config(self) -> "PretrainedConfig":
+        raise NotImplementedError
+
+    @property
+    def tron_config(self) -> GPTConfig | T5Config:
+        if self._tron_config is None:
+            raise ValueError("Tron config is not set")
+        return self._tron_config
+
+    @property
+    def tokenizer(self) -> "AutoTokenizer":
+        if self._tokenizer is None:
+            if self._hf_tokenizer_path is not None:
+                from transformers import AutoTokenizer
+
+                self._tokenizer = AutoTokenizer.from_pretrained(self._hf_tokenizer_path, trust_remote_code=True)
+
+        return self._tokenizer
+
+    def convert_state(self, source, target):
+        raise NotImplementedError
+
+    def init_hf_model(self, dtype=torch.bfloat16) -> "AutoModelForCausalLM":
+        """Initialize a new Hugging Face Llama model with the specified data type.
+
+        Args:
+            dtype (torch.dtype): The data type for the model parameters. Default: torch.bfloat16
+
+        Returns:
+            LlamaForCausalLM: An initialized Hugging Face Llama model with no weights loaded
+        """
+        from transformers import AutoModelForCausalLM
+        from transformers.modeling_utils import no_init_weights
+
+        with no_init_weights(True):
+            return AutoModelForCausalLM.from_config(self.hf_config, torch_dtype=dtype)
+
+    def init_tron_model(self) -> tuple[dict, dict]:
+        """
+        This function loads the state dict directly from a distributed checkpoint, and modify the state dict
+        so that it is consistent with the key names you would get from loading the checkpoint into a model.
+        This is a more memory-efficient method to obtain a state dict without initializing the nemo model.
+
+        Args:
+            path (Path): The path from which the model will be loaded.
+
+        Returns
+        -------
+            tuple[Dict, Dict]: The loaded state dict and the yaml config dict.
+        """
+        tron_yaml = self.input_path / "run_config.yaml"
+        assert tron_yaml.exists(), f"Tron config file {tron_yaml} does not exist"
+        with open(tron_yaml, "r") as stream:
+            _config = yaml.safe_load(stream)
+        config = _config["model_config"]
+        config = instantiate(config)
```

Comment:
easy to mix up `_config` and `config` in this method, especially below when the tokenizer config is being read

```suggestion
        model_config = _config["model_config"]
        model_config = instantiate(model_config)
```